scripts/sct_label_utils.py

```diff
@@ -681,195 +683,243 @@ def get_parser():
-                      type_value='file',
-                      description='Compute Mean Square Error between labels from input and reference image. Specify reference image here.',
-                      mandatory=False)
-    parser.add_option(name='-remove-reference',
-                      type_value='file',
-                      description='Remove labels from input image (-i) that are not in reference image (specified here).',
-                      mandatory=False)
-    parser.add_option(name='-remove-sym',
-                      type_value='file',
-                      description='Remove labels from input image (-i) and reference image (specified here) that don\'t match. You must provide two output names separated by ",".',
-                      mandatory=False)
-    parser.add_option(name='-remove',
-                      type_value=[[','], 'int'],
-                      description='Remove labels of specific value (specified here) from reference image',
-                      mandatory=False)
-    parser.add_option(name='-keep',
-                      type_value=[[','], 'int'],
-                      description='Keep labels of specific value (specified here) from reference image',
-                      mandatory=False)
-
-    parser.usage.addSection("MISC")
-    parser.add_option(name="-msg",
-                      type_value="str",
-                      description='Display a message to explain the labeling task. Use with -create-viewer.',
-                      mandatory=False)
-    parser.add_option(name="-o",
-                      type_value=[[','], "file_output"],
-                      description="Output image(s).",
-                      mandatory=False,
-                      example="t2_labels_cross.nii.gz",
-                      default_value="labels.nii.gz")
-    parser.add_option(name="-v",
-                      type_value="multiple_choice",
-                      description='Verbose. 0: nothing. 1: basic. 2: extended.',
-                      mandatory=False,
-                      default_value=param_default.verbose,
-                      example=['0', '1', '2'])
-    parser.add_option(name='-qc',
-                      type_value='folder_creation',
-                      description='The path where the quality control generated content will be saved',
-                      default_value=None)
-    parser.add_option(name='-qc-dataset',
-                      type_value='str',
-                      description='If provided, this string will be mentioned in the QC report as the dataset the process was run on',
-                      )
-    parser.add_option(name='-qc-subject',
-                      type_value='str',
-                      description='If provided, this string will be mentioned in the QC report as the subject the process was run on',
-                      )
+    parser = argparse.ArgumentParser(
+        description="Utility function for label image.",
+        formatter_class=SmartFormatter,
```

Comment:
We have this everywhere, maybe not worth the added mental overhead but we could have and `sct_parser` function that wraps `argparse.ArgumentParser` and sets this at the default

Reply:
You're right, that same pattern is used everywhere, and it is a bit repetitive.

On the other hand, this way of doing it nice because what exactly is being set is clear at the location the parser is defined. If we used `sct_parser`, it wouldn't be immediately obvious what's different about it from the normal parser, so you'd have to go to the source to find out: "Ah! `sct_parser` just sets `formatter_class`."

Reply:
In a theoretical `sct_parser`, there are other defaults that could be set, too. e.g. we could have common arguments like `verbose` and `remove_temp` (`-v` and `-r`) added there too.

Reply:
> so you'd have to go to the source to find out: "Ah! sct_parser just sets formatter_class but nothing else."

This is what I meant by overhead. I think the repeating this argument is probably preferable, but you are right there are more common arguments we could be setting. If we chose to go with a custom class we could have `optional` and `required` as part of the parser instead of separate groups, so we could do e.g. `parser.optional.add_argument`

Reply:
Resolving, because this is good for now (possibly forever)